test/Microsoft.AspNet.Mvc.Core.Test/OptionDescriptors/OptionDescriptorExtensionsTest.cs

```diff
@@ -0,0 +1,276 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.OptionDescriptors
+{
+    public class OptionDescriptorExtensionsTest
```

Comment:
FYI one pattern for testing base classes that might have worked here is
``` c#
public abstract class OptionDescriptorExtensionsTestBase<TOption>
{
  protected abstract List<IOptionDescriptor<TOption>> GetDescriptors(MvcOptions options);
  protected abstract TOption GetInstance(); // return new instance on each call
  protected abstract TOption GetMockInstance(); // return new mock instance on each call
  ...
}
...
public class ExcludeValidationDescriptorTests : OptionDescriptorExtensionsTestBase<IExcludeTypeValidationFilter>
{
  ...
}
```
but I suspect in this case it would get fairly complex and covariance might be a blocker.